app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt

```diff
@@ -118,5 +119,32 @@ object ThreadController {
     fun deleteThreads(folderId: String, realm: MutableRealm) {
         realm.delete(getThreadsQuery(folderId, realm = realm))
     }
+
+    fun fetchIncompleteMessages(thread: Thread) {
+        RealmDatabase.mailboxContent().writeBlocking {
+            thread.messages.forEach { localMessage ->
+                if (!localMessage.fullyDownloaded) {
+                    ApiRepository.getMessage(localMessage.resource).data?.also { remoteMessage ->
```

Comment:
No API calls in Controllers.

Reply:
In this case, we can leave it as it is realm that drives the api call